### PR TITLE
Add pygeoapi

### PR DESCRIPTION
**Pull request by hwbllmnn**
- add 2 processes:
  - Zonal Stats with GDAL
  - Zonal Stats with GRASS

### CODE REVIEW

**Comment by hwbllmnn** [comment]
Note that the current code is not working yet.

File: pygeoapi/processes/zonal_statistics.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        i = 0
+        result = None
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            with open(f'{tempdir}/output', 'w') as output:
+                subprocess.run(['grass', '-c', 'EPSG:4326', f'{tempdir}/grass', '-e'])
+                session = gsetup.init(f'{tempdir}/grass/')
+
+                r.in_gdal(input = '/opt/geoserver_data/sample.tif', output = 'raster')
+                for geom in geoms:
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics.py:69]: we could use `enumarate()` to automatically create the index count. see https://www.geeksforgeeks.org/enumerate-in-python/

```python
for i, geom in enumerate(geoms):
  print(i, geom )
```

Comment (JakobMiksch) [reply pygeoapi/processes/zonal_statistics.py]: done

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+                demo_raster_name = 'demo-raster'
+                # input raster must be in EPSG 4326
+                r.in_gdal(input='/demo_data/ecostress_4326.tif',
+                          output=demo_raster_name)
+
+                # set region
+                g.region(raster=demo_raster_name)
+
+                # TODO: use r.external instead of r.gdal_in
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:79]: @hblitza I did not make it anymore to implement it

**Pull request merged**